vm_supervisor/storage.py

```diff
@@ -155,28 +162,100 @@ def create_ext4(path: Path, size_mib: int) -> bool:
     return True
 
 
+async def create_devmapper(volume: Union[PersistentVolume, RootfsVolume], namespace: str) -> Path:
+    """It creates a /dev/mapper/DEVICE inside the VM, that is an extended mapped device of the volume specified.
+    We follow the steps described here: https://community.aleph.im/t/deploying-mutable-vm-instances-on-aleph/56/2"""
+    volume_name = volume.name if isinstance(volume, PersistentVolume) else "rootfs"
+    mapped_volume_name = f"{namespace}_{volume_name}"
+    path_mapped_volume_name = Path("/dev/mapper") / mapped_volume_name
+
+    if path_mapped_volume_name.is_block_device():
+        # If the target block device is already set return it
+        return path_mapped_volume_name
+
+    path = Path(settings.PERSISTENT_VOLUMES_DIR) / namespace / f"{volume_name}.ext4"
+    if not path.is_file():
+        # If the target file don´t exists, create it
+        logger.debug(f"Creating {volume.size_mib}MB volume")
+        os.system(f"dd if=/dev/zero of={path} bs=1M count={volume.size_mib}")
+        if settings.USE_JAILER:
+            os.system(f"chown jailman:jailman {path}")
+
+    parent_path = await get_runtime_path(volume.parent.ref)
+    # Create two loopback devices one as a base and the second one as an extended volume
+    base_loop_device = subprocess.run(
+        ["losetup", "--find", "--show", "--read-only", parent_path],
+        check=True,
+        capture_output=True,
+        encoding="UTF-8").stdout.strip()
+    base_block_size = subprocess.run(
+        ["blockdev", "--getsz", parent_path],
+        check=True,
+        capture_output=True,
+        encoding="UTF-8").stdout.strip()
+    extended_loop_device = subprocess.run(
+        ["losetup", "--find", "--show", path],
+        check=True,
+        capture_output=True,
+        encoding="UTF-8").stdout.strip()
+    extended_block_size = subprocess.run(
+        ["blockdev", "--getsz", path],
+        check=True,
+        capture_output=True,
+        encoding="UTF-8").stdout.strip()
+    # Creates the first mapped device using the base disk image
+    table_command = f"0 {base_block_size} linear {base_loop_device} 0\n{base_block_size} {extended_block_size} zero"
+    base_volume_name = f"{namespace}_{volume_name}_base"
+    os.system(f"printf \"{table_command}\" | dmsetup create {base_volume_name}")
```

Comment:
Can we avoid using `os.system` ?

```suggestion
    subprocess.run(cmd=f"dmsetup create {base_volume_name}", input=table_command, text=True, shell=True, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
)
```